File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/CommunicationRequest.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class CommunicationRequest : DomainResource, IResource
{
    public Base.CodeableConcept[]? Category { get; set; }
    public CommunicationRequest_Payload[]? Payload { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public Base.CodeableConcept[]? Medium { get; set; }
    public Base.ResourceReference[]? Recipient { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public Base.CodeableConcept? StatusReason { get; set; }
    public string? AuthoredOn { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public Base.ResourceReference? Requester { get; set; }
    public string? Priority { get; set; }
    public Base.Period? OccurrencePeriod { get; set; }
    public required string Status { get; set; }
    public Base.Identifier? GroupIdentifier { get; set; }
    public Base.ResourceReference? Sender { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public bool? DoNotPerform { get; set; }
    public Base.ResourceReference[]? Replaces { get; set; }
    public Base.ResourceReference[]? BasedOn { get; set; }
    public string? OccurrenceDateTime { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public object? Occurrence    
    {
        get
        {
            if (OccurrencePeriod is not null)
            {
                return OccurrencePeriod;
            }
    
            if (OccurrenceDateTime is not null)
            {
                return OccurrenceDateTime;
            }
    
            return null;
        }
    
        set
        {
            if (value?.GetType() == typeof(Base.Period))
            {
                OccurrencePeriod = (Base.Period)value;return;
            }
    
            if (value?.GetType() == typeof(string))
            {
                OccurrenceDateTime = (string)value;return;
            }
    
            throw new ArgumentException("Invalid type provided");
        }
    }
    public Base.ResourceReference[]? About { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class CommunicationRequest_Payload : BackboneElement
    {
        public object? Content    
        {
            get
            {
                if (ContentString is not null)
                {
                    return ContentString;
                }
        
                if (ContentReference is not null)
                {
                    return ContentReference;
                }
        
                if (ContentAttachment is not null)
                {
                    return ContentAttachment;
                }
        
                return null;
            }
        
            set
            {
                if (value?.GetType() == typeof(string))
                {
                    ContentString = (string)value;return;
                }
        
                if (value?.GetType() == typeof(Base.ResourceReference))
                {
                    ContentReference = (Base.ResourceReference)value;return;
                }
        
                if (value?.GetType() == typeof(Base.Attachment))
                {
                    ContentAttachment = (Base.Attachment)value;return;
                }
        
                throw new ArgumentException("Invalid type provided");
            }
        }
        public string? ContentString { get; set; }
        public Base.ResourceReference? ContentReference { get; set; }
        public Base.Attachment? ContentAttachment { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/CommunicationRequest.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Attachment
from base import Period
from base import CodeableConcept
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class CommunicationRequestPayload(BackboneElement):
    content_string: Optional[str] = None
    content_reference: Optional[Reference] = None
    content_attachment: Optional[Attachment] = None

@dataclass(kw_only=True)
class CommunicationRequest(DomainResource):
    category: Optional[List[CodeableConcept]] = field(default_factory=list)
    payload: Optional[List[CommunicationRequestPayload]] = field(default_factory=list)
    encounter: Optional[Reference] = None
    medium: Optional[List[CodeableConcept]] = field(default_factory=list)
    recipient: Optional[List[Reference]] = field(default_factory=list)
    reason_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    status_reason: Optional[CodeableConcept] = None
    authored_on: Optional[str] = None
    note: Optional[List[Annotation]] = field(default_factory=list)
    requester: Optional[Reference] = None
    priority: Optional[str] = None
    occurrence_period: Optional[Period] = None
    status: str
    group_identifier: Optional[Identifier] = None
    sender: Optional[Reference] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    do_not_perform: Optional[bool] = None
    replaces: Optional[List[Reference]] = field(default_factory=list)
    based_on: Optional[List[Reference]] = field(default_factory=list)
    occurrence_date_time: Optional[str] = None
    subject: Optional[Reference] = None
    about: Optional[List[Reference]] = field(default_factory=list)
    reason_reference: Optional[List[Reference]] = field(default_factory=list)